src/ecs.rs

```diff
@@ -0,0 +1,692 @@
+    /// - admin
+    /// - allowed
+    /// - change
+    /// - connection
+    /// - creation
+    /// - deletion
+    /// - denied
+    /// - end
+    /// - error
+    /// - group
+    /// - indicator
+    /// - info
+    /// - installation
+    /// - protocol
+    /// - start
+    /// - user
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        rename(serialize = "type", deserialize = "type")
+    )]
+    type_: Option<Vec<String>>,
+    /// Outcome simply denotes whether the event represents a success or a failure from the perspective of
+    /// the entity that produced the event. It is the lowest-level event categorization in ECS.
+    ///
+    /// # Allowed values
+    ///
+    /// - failure
+    /// - success
+    /// - unknown
+    #[serde(skip_serializing_if = "Option::is_none")]
+    outcome: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsLogFieldSet {
+    /// Original log level of the log event.
+    level: String,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsHttpFieldSet {
+    /// Fields related to an HTTP request.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    request: Option<EcsHttpRequestFieldSet>,
+    /// Fields related to an HTTP response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    response: Option<EcsHttpResponseFieldSet>,
+    /// HTTP version.
+    ///
+    /// # Example
+    ///
+    /// `1.1`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    version: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsHttpRequestFieldSet {
+    /// HTTP request method.
+    ///
+    /// # Example
+    ///
+    /// `POST`
+    method: String,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsHttpResponseFieldSet {
+    /// HTTP status code.
+    ///
+    /// # Example
+    ///
+    /// `404`
+    status: i64,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsUrlFieldSet {
+    /// The original URL value, as processed.
+    original: String,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsUserAgentFieldSet {
+    /// The original user agent value, as processed.
+    original: String,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsRiskFieldSet {
+    /// The risk classification calculated by the system.
+    calculated_level: String,
+    /// The risk score calculated by the system.
+    calculated_risk_score: f64,
+    /// The risk score calculated by the system, after normalization.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    calculated_risk_score_norm: Option<f64>,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsFileFieldSet {
+    /// The full path to the file.
+    ///
+    /// # Example
+    ///
+    /// `/home/alice/plugins/example.wasm`
```

Comment:
Currently the only event that uses this is the plugin load event, which will be the full path to the plugin file. However, permissioned file access is a roadmap item and it may make use of this too.